frontend/src/features/admin-form/settings/SettingsPage.tsx

```diff
@@ -48,14 +40,32 @@ export const SettingsPage = (): JSX.Element => {
 
   const { ref, onMouseDown } = useDraggable<HTMLDivElement>()
 
-  const [tabIndex, setTabIndex] = useState(
-    settingsTabsOrder.indexOf(settingsTab ?? ''),
+  // Note: Admins are not redirected to /general on invalid settings tabs as we
+  // don't want to do this prematurely before displayPayments can be determined.
+  const tabConfig = useMemo(() => {
+    const baseTabs = [
+      { label: 'General', icon: BiCog, component: SettingsGeneralPage },
+      { label: 'Singpass', icon: BiKey, component: SettingsAuthPage },
+      {
+        label: 'Twilio credentials',
+        icon: BiMessage,
+        component: SettingsTwilioPage,
+      },
+      { label: 'Webhooks', icon: BiCodeBlock, component: SettingsWebhooksPage },
+      { label: 'Payments', icon: BiDollar, component: SettingsPaymentsPage },
```

Comment:
Might want to separate the `label` from navigation path. There is no guarantee that the label can be represented as a readable slug, i.e., for labels with multiple words `Email Notification` may be displayed as `email%20notification` in the url

Reply:
That's a good point, will add a separate var for `path` in the `tabConfig`. Thanks!